ecs-task-definition-deploy/action.yml

```
@@ -0,0 +1,123 @@
+
+    - name: Setup Python + Poetry
+      uses: moneymeets/action-setup-python-poetry@master
+      with:
+        working_directory: ${{ github.action_path }}
+        # ToDo: Re-enable cache when https://github.com/actions/setup-python/issues/361 is fixed
+        poetry_cache_enabled: 'false'
+
+    - name: Get latest active task definition created by Pulumi
+      id: get-pulumi-task-definition-arn
+      shell: bash
+      env:
+        APPLICATION_ID: ${{ inputs.application_id }}
+      run: |
+        TASK_DEFINITION_ARN=$(
+          poetry run --directory ${{ github.action_path }} actions_helper get-active-task-definition-arn-by-tag \
+            --application-id "$APPLICATION_ID" \
+            --tags "created_by:Pulumi, Name:$APPLICATION_ID"
+        )
+        echo "task-definition-arn=${TASK_DEFINITION_ARN}" >> $GITHUB_OUTPUT
+
+    - name: Prepare task-definition.json based on active Pulumi latest
+      shell: bash
+      run: |
+        aws ecs describe-task-definition \
+          --task-definition ${{ steps.get-pulumi-task-definition-arn.outputs.task-definition-arn }} \
+          --query taskDefinition \
+          --region ${{ inputs.aws_region }} \
+        > task-definition.json
+
+        result=$(jq '.' task-definition.json | jq '.containerDefinitions[].image | select(. != "PLACEHOLDER")')
+        if [ "$result" ]; then
+            echo "Error: Not all values for containerDefinitions 'image' equal to 'PLACEHOLDER'"
+            exit 1
+        fi
+
+    - name: Replace PLACEHOLDER in task definitions with Image URI
+      shell: bash
+      run: |
+        jq --arg image_uri "${{ inputs.image_uri }}" '.containerDefinitions[].image |= $image_uri' \
+          task-definition.json > tmpfile && mv tmpfile task-definition.json
+    
+    - name: Get latest active task definition created by GitHub Actions Deployment
+      id: get-github-action-task-definition-arn
+      shell: bash
+      env:
+        APPLICATION_ID: ${{ inputs.application_id }}
+      run: |
+        TASK_DEFINITION_ARN=$(
+          poetry run --directory ${{ github.action_path }} actions_helper get-active-task-definition-arn-by-tag \
+            --application-id "$APPLICATION_ID" \
+            --tags "created_by:GitHub Actions Deployment, Name:$APPLICATION_ID" \
+            --allow-empty
+        )
+        echo "task-definition-arn=${TASK_DEFINITION_ARN}" >> $GITHUB_OUTPUT
+    
+    - name: Deregister previous task definition
+      shell: bash
+      if: inputs.deregister_previous == 'true' && steps.get-github-action-task-definition-arn.outputs.task-definition-arn != ''
+      run: |
+        aws ecs deregister-task-definition \
+          --task-definition ${{ steps.get-github-action-task-definition-arn.outputs.task-definition-arn }} \
+          --region ${{ inputs.aws_region }}
+
+    - name: Deploy task definition to Amazon ECS
+      id: deploy-task-definition
+      uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+      with:
```

Some time ago, the team has decided to only use name for steps when it's not immediately clear for the command or action used in the step. Here, I think it's clear, so we should remove the name.
```suggestion
    - uses: aws-actions/amazon-ecs-deploy-task-definition@v1
      id: deploy-task-definition      
      with:
```